### PR TITLE
specify namespace for component secret

### PR DESCRIPTION
**Pull request by philbrookes**
tell the backup job which namespace the component credentials secret will be stored in.

### CODE REVIEW

**Comment by pb82** [comment]
:eyes:

**Comment by pb82** [comment]
@philbrookes the backup job is created but currently fails and i believe it's because the operator is still trying to mount the aws and encryption secrets: https://github.com/integr8ly/keycloak-operator/blob/master/pkg/keycloak/phaseHandler.go#L235 This would only work if they were in the same namespace but they are in the default namespace now.

We removed this in the templates and only expose the names of the secrets as env vars: https://github.com/integr8ly/backup-container-image/blob/master/templates/openshift/backup-cronjob-template.yaml#L38 and reference them using oc: https://github.com/integr8ly/backup-container-image/blob/master/image/tools/lib/component/postgres.sh#L3

**Comment by philbrookes** [comment]
@pb82 I've addressed that issue now, if you could take another look

File: pkg/keycloak/phaseHandler.go [hunk 1]
```diff
@@ -231,21 +231,19 @@ func (ph *phaseHandler) reconcileBackup(sso *v1alpha1.Keycloak, backup v1alpha1.
 								{
 									Name:    backup.Name + "-keycloak-backup",
 									Image:   backup.Image + ":" + backup.ImageTag,
-									Command: []string{"/opt/intly/tools/entrypoint.sh", "-c", "postgres", "-b", "s3", "-e", "gpg"},
-									EnvFrom: []v1.EnvFromSource{
+									Command: []string{"/opt/intly/tools/entrypoint.sh", "-c", "postgres", "-n", namespace, "-b", "s3", "-e", "gpg"},
```
Comment (pb82) [inline pkg/keycloak/phaseHandler.go]: Do you need to add the new `n` flag here to set the component secret namespace?

**Comment by pb82** [comment]
@philbrookes we need to set the serviceaccount to `backupjob` for the cronjob here: https://github.com/integr8ly/keycloak-operator/blob/37876baa23c99fd0707384d3a50b9c1af27c3456/pkg/keycloak/phaseHandler.go#L230

it should look like this:
```go
						Spec: v1.PodSpec{
							ServiceAccountName: "backupjob",
							Containers: []v1.Container{
								{
```

**Comment by philbrookes** [comment]
@pb82 I've added that serviceaccount to the template now.

**Pull request merged**